### ISSUE TITLE
Cannot save updates to user-defined variables in the Variables document

### ISSUE THREAD

**Issue by a-s-russo** [jdemetra/jdplus-main#385]
When I make changes to user-defined variables in the Variables document of my workspace, certain changes such as inserting new ones or removing existing ones, will not be saved. In one instance, when I click on 'Save Workspace', the underlying XML file is not updated, and in another instance, I cannot even select 'Save Workspace' as it is greyed-out. The only solution to save my changes appears to be selecting 'Save Workspace As...' instead, which is highly undesirable.

![image](https://github.com/user-attachments/assets/3ca10c7a-d5d3-4e7e-bc4e-df79a0121892)

It will be tedious for me to create a video to demonstrate the bug. I suggest I meet with one of the developers and share screens to explain the issue if necessary.

**Comment by a-s-russo** [comment]
FYI @Immurb I added some videos to this issue to explain the problem further.

**Comment by Immurb** [comment]
Yeah this is definitely a bug. The internal classes change their state successfully, but don't communicate it to the UI elements.

**Comment by charphi** [comment]
Fixed in 3.5.0
